pkg/signing/handlers/sigstore/handler.go

```diff
@@ -0,0 +1,242 @@
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package sigstore
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/fulcio"
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/v2/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	"github.com/sigstore/rekor/pkg/types/rekord"
+	rekorv001 "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"github.com/sigstore/rekor/pkg/verify"
+	"github.com/sigstore/sigstore/pkg/signature"
+
+	"github.com/open-component-model/ocm/pkg/contexts/credentials"
+	"github.com/open-component-model/ocm/pkg/errors"
+	"github.com/open-component-model/ocm/pkg/signing"
+)
+
+// Algorithm defines the type for the RSA PKCS #1 v1.5 signature algorithm.
+const Algorithm = "sigstore"
+
+// MediaType defines the media type for a plain RSA signature.
+const MediaType = "application/vnd.ocm.signature.sigstore"
+
+// SignaturePEMBlockAlgorithmHeader defines the header in a signature pem block where the signature algorithm is defined.
+const SignaturePEMBlockAlgorithmHeader = "Algorithm"
+
+func init() {
+	signing.DefaultHandlerRegistry().RegisterSigner(Algorithm, Handler{})
+}
+
+// Handler is a signatures.Signer compatible struct to sign using sigstore
+// and a signatures.Verifier compatible struct to verify using sigstore.
+type Handler struct{}
+
+func (h Handler) Algorithm() string {
+	return Algorithm
+}
+
+func (h Handler) Sign(cctx credentials.Context, digest string, hash crypto.Hash, issuer string, key interface{}) (*signing.Signature, error) {
+	ctx := context.Background()
+
+	priv, err := cosign.GeneratePrivateKey()
+	if err != nil {
+		return nil, fmt.Errorf("error generating keypair: %w", err)
+	}
+
+	signer, err := signature.LoadECDSASignerVerifier(priv, hash)
+	if err != nil {
+		return nil, fmt.Errorf("error loading sigstore signer: %w", err)
+	}
+
+	fs, err := fulcio.NewSigner(ctx, options.KeyOpts{
+		FulcioURL:        "https://v1.fulcio.sigstore.dev",
+		OIDCIssuer:       "https://oauth2.sigstore.dev/auth",
+		OIDCClientID:     "sigstore",
+		SkipConfirmation: true,
+	}, signer)
+	if err != nil {
+		return nil, errors.Wrap(err, "new signer")
+	}
+
+	sig, err := fs.SignMessage(strings.NewReader(digest))
```

Comment:
I think here a digest from the digest is calculated. Is this a good idea?

Reply:
Yes that's correct. The input digest is a representation of the content so I don't think it's an issue.  

The other option here is to directly interact with the Fulcio server to request signing certificates and use those to sign the digest.